src/utils/DockerUtil.js

```diff
@@ -330,6 +332,85 @@ export default {
     });
   },
 
+  active (name) {
+    this.detach();
+    this.activeContainerName = name;
+
+    if (name) {
+      this.logs();
+    }
+  },
+
+  logs () {
+    if (!this.activeContainerName) {
+      return;
+    }
+
+    this.client.getContainer(this.activeContainerName).logs({
+      stdout: true,
+      stderr: true,
+      tail: 1000,
+      follow: false,
+      timestamps: 1
+    }, (err, logStream) => {
+      if (err) {
+        return;
+      }
+
+      let logs = '';
+      logStream.setEncoding('utf8');
+      logStream.on('data', chunk => logs += chunk);
+      logStream.on('end', () => {
+        containerServerActions.logs({name: this.activeContainerName, logs});
+        this.attach();
+      });
+    });
+  },
+
+  attach () {
```

Comment:
They are similar in that they both call the /logs api endpoint, but they behave differently - one streams changes and the other hooks into the logs. I tried merging them in various ways, but kept leaning towards two functions.